### PR TITLE
infra: add log rotation support (containers)

### PR DESCRIPTION
**Pull request by guits**
This commit adds the log rotation support via logrotate in containerized
deployments.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1848388

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

File: roles/ceph-infra/templates/logrotate.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    rotate 7
+    daily
+    compress
+    sharedscripts
+    postrotate
+        killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror || pkill -1 -x "ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror" || true
+    endscript
+    missingok
+    notifempty
+    su root root
```
Comment (guits) [inline roles/ceph-infra/templates/logrotate.conf.j2:11]: NOTE to myself:
must figure something about this

**Comment by guits** [comment]
jenkins test centos-non_container-purge

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
jenkins test centos-non_container-purge

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-purge

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
jenkins test centos-non_container-update

**Comment by guits** [comment]
jenkins test centos-non_container-purge

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-update

File: roles/ceph-infra/templates/logrotate.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/var/log/ceph/*.log {
+    rotate 7
+    daily
+    compress
+    sharedscripts
+    postrotate
+        killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror || pkill -1 -x "ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror" || true
```
Comment (rmahique) [inline roles/ceph-infra/templates/logrotate.conf.j2:7]: Is this necessary? i don't see any process taking hold of the log files.

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
jenkins test centos-non_container-purge

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

File: roles/ceph-infra/templates/logrotate.conf.j2 (hunk 2 above)
Comment (guits) [reply roles/ceph-infra/templates/logrotate.conf.j2:7]: trying to keep consistency with what is made for RPM based deployment https://github.com/ceph/ceph/blob/master/src/logrotate.conf

not sure it is really needed but it doesn't hurt anyway.

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Pull request merged**